internal/core/CMakeLists.txt

```diff
@@ -133,9 +133,25 @@ if ( APPLE )
             "-Wno-unused-parameter"
             "-Wno-deprecated"
             "-DBOOST_STACKTRACE_GNU_SOURCE_NOT_REQUIRED=1"
+            #"-fvisibility=hidden"
+            #"-fvisibility-inlines-hidden"
             )
 endif ()
 
+# Set SIMD to CMAKE_CXX_FLAGS
+message(STATUS "CPU_ARCH:${CPU_ARCH}")
+if (${CPU_ARCH} STREQUAL "avx")
+  #set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}  -ftree-vectorize -mavx2 -mfma -mavx -mf16c ")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mavx2 -mfma -mavx -mf16c ")
+elseif (${CPU_ARCH} STREQUAL "sse")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -msse4.2 ")
+elseif (${CPU_ARCH} STREQUAL "aarch64")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mcpu=neoverse-n1 ")
+elseif (${CPU_ARCH} STREQUAL "arm64")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mcpu=apple-m1+crc ")
+endif()
```

Comment:
To my understanding. This asks compiler optimize SMID, but its pre-assumption is that the building host must have the same CPU arch as the running host. Or Milvus cannot run.

Reply:
Yes. this is a point

Reply:
this may course binary not work on some old cpu. aarch64 not only neoverse-n1.

Reply:
we don't use these flag default, because -O3 can use SSE2 simd optimization and get a good performance and SSE2 is base line for x86 machine.  For developer who compile milvus by himself, he can use high level simd like mavx2 and other.